### PR TITLE
Update network-storage interface to support runners and pass daos scripts during startup

### PR DESCRIPTION
**Pull request by harshthakkar01**
Currently, network-storage interface doesn't support client-install and mount runner scripts and we don't need that for natively supported file-systems and also these file-systems are mounted using fstab. For parallelstore we need to pass client-install script and mount scripts separately since it needs to install drivers and it uses dfuse to mount container. 

This PR adds support for client-install and mount-install runner in network-storage interface and only run the scripts for DAOS file-system. 

For,
* DAOS file system Update startup scripts to run client-install and mount runners and don't update config.yaml file. 

### Submission Checklist

Please take the following actions before submitting this pull request.

* Fork your PR branch from the Toolkit "develop" branch (not main)
* Test all changes with pre-commit in a local branch [#](https://goo.gle/hpc-toolkit#development)
* Confirm that "make tests" passes all tests
* Add or modify unit tests to cover code changes
* Ensure that unit test coverage remains above 80%
* Update all applicable documentation
* Follow Cloud HPC Toolkit Contribution guidelines [#](https://goo.gle/hpc-toolkit-contributing)

### CODE REVIEW

File: community/modules/scheduler/schedmd-slurm-gcp-v6-controller/slurm_files.tf [hunk 1]
```diff
@@ -69,23 +69,72 @@ resource "google_storage_bucket_iam_binding" "legacy_readers" {
...
-    filename = "ghpc_startup.sh"
-    content  = var.login_startup_script
+  ghpc_startup_script_controller = [
+    {
+      filename = "ghpc_startup.sh"
+      content  = var.controller_startup_script
+    },
+    {
+      filename = "daos_mount.sh"
+      content  = local.daos_install_mount_script
```
[the remark this answers is not in the document]
Comment (mr0re1) [reply community/modules/scheduler/schedmd-slurm-gcp-v6-controller/slurm_files.tf:105]: This is somewhat expensive to always create `module.daos_network_storage_scripts`
Did you consider to just add them to `ghpc_startup_script_controller`:

```hcl
daos_scripts = [{filename: "ghpc_daos_${i}", content: v.content} for i,v in local.daos_network_storage_runners]
...
ghpc_startup_script_controller = concat([{
   filename = "ghpc_startup.sh"
   content  = var.controller_startup_script
}], local.daos_scripts)

```

**Review by mr0re1 — APPROVED** [review]

File: community/modules/scheduler/schedmd-slurm-gcp-v6-controller/slurm_files.tf [hunk 2]
```diff
@@ -69,23 +69,72 @@ resource "google_storage_bucket_iam_binding" "legacy_readers" {
...
-  ghpc_startup_script_compute = [{
-    filename = "ghpc_startup.sh"
-    content  = var.compute_startup_script
+  ghpc_startup_script_login = [
+    {
+      filename = "ghpc_startup.sh"
+      content  = var.login_startup_script
+    },
+    {
+      filename = "daos_mount.sh"
```
Comment (harshthakkar01) [inline community/modules/scheduler/schedmd-slurm-gcp-v6-controller/slurm_files.tf:113]: Run this before ghpc-startup script.

File: community/modules/scheduler/schedmd-slurm-gcp-v6-controller/slurm_files.tf [hunk 3]
```diff
@@ -121,8 +170,12 @@ module "slurm_files" {
   enable_slurm_gcp_plugins      = var.enable_slurm_gcp_plugins
 
   disable_default_mounts = !var.enable_default_mounts
-  network_storage        = var.network_storage
-  login_network_storage  = var.login_network_storage
+
+  network_storage = [
+    for storage in var.network_storage : storage
+    if storage.fs_type != "daos"
```
Comment (harshthakkar01) [inline community/modules/scheduler/schedmd-slurm-gcp-v6-controller/slurm_files.tf:176]: Make sure we don't pass scripts for file-systems other than daos.

**Comment by harshthakkar01** [comment]
LGTMed By Ivan.

**Pull request merged**